src/controlplane/client/connector.go

```diff
@@ -0,0 +1,294 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+)
+
+const (
+	DefaultTimout      = 5000 * time.Millisecond
+	defaultMetricsPath = "/metrics"
+	mTLSAuth           = "mTLS"
+	bearerAuth         = "bearer"
+)
+
+// Connector provides an interface to retrieve []connParams to connect to a Control Plane instance.
+type Connector interface {
+	Connect() (*connParams, error)
+}
+
+type defaultConnector struct {
+	// TODO: Use a non-sdk logger
+	logger          log.Logger
+	kc              kubernetes.Interface
+	inClusterConfig *rest.Config
+	endpoints       []config.Endpoint
+}
+
+// DefaultConnector returns a defaultConnector that probes all endpoints in the list and return the first responding status OK.
+func DefaultConnector(endpoints []config.Endpoint, kc kubernetes.Interface, inClusterConfig *rest.Config, logger log.Logger) (Connector, error) {
+	if inClusterConfig == nil {
+		return nil, fmt.Errorf("inClusterConfig cannot be nil")
+	}
+
+	if logger == nil {
+		return nil, fmt.Errorf("logger cannot be nil")
+	}
+
+	if kc == nil {
+		return nil, fmt.Errorf("kubernetes interface cannot be nil")
+	}
+
+	if len(endpoints) == 0 {
+		return nil, fmt.Errorf("endpoints cannot be empty")
+	}
+
+	if err := validateEndpointConfig(endpoints); err != nil {
+		return nil, fmt.Errorf("validating endpoints config: %w", err)
+	}
+
+	return &defaultConnector{
+		logger:          logger,
+		inClusterConfig: inClusterConfig,
+		kc:              kc,
+		endpoints:       endpoints,
+	}, nil
+}
+
+// Connect iterates over the endpoints list probing each endpoint with a HEAD request
+// and returns the connection parameters of the first endpoint that respond Status OK.
+func (dp *defaultConnector) Connect() (*connParams, error) {
+	for _, e := range dp.endpoints {
+		dp.logger.Debugf("Configuring autodiscover endpoint %q for probing", e.URL)
+
+		u, err := url.Parse(e.URL)
+		if err != nil {
+			return nil, fmt.Errorf("parsing endpoint url %q: %w", e.URL, err)
+		}
+
+		// If no path is defined on the config, default is set.
+		if u.Path == "" || u.Path == "/" {
```

Comment:
```suggestion
		if strings.TrimSuffix(u.Path, "/") == "" {
```

So we don't mess up if path is `http://localhost///`

Reply:
On a second thought, I do wonder what happens if I want to point the integration to a custom endpoint that _does_ serve metrics on `/` 🤔